platform/options.keymap/src/org/netbeans/modules/options/keymap/KeymapPanel.java

```diff
@@ -299,8 +299,9 @@ private void narrowByShortcut() {
             getModel().runWithoutEvents(new Runnable() {
                 public void run() {
                     getModel().getDataVector().removeAllElements();
-                    for (String categorySet : getModel().getCategories().keySet()) {
-                        for (String category : getModel().getCategories().get(categorySet)) {
+
+                    for (Map.Entry<String, List<String>> entry : getModel().getCategories().entrySet()) {
+                        for (String category : entry.getValue()) {
```

Comment:
Almost done, I'm getting punchy... Again not looking for a change, just mentioning, a check on myself

```
for(List<String> categories : getModel().getCategories().values()) {
    for(String category : categories) {
```